### PR TITLE
@joystream/types - checks config, CI, linter fixes

### PR DESCRIPTION
**Pull request by Lezek123**
This PR:
- Adds yarn `lint`, `format` and `checks` (which also includes circular dep. check) scripts for `@joystream/types`
- Configures linter in the `@joystream/types` workspace and turns off some case-related rules that would require breaking changes
-  Fixes the remaining linter issues
- Adds CI checks specifically for `@joystream/types` and removes no-longer neccessary `madge --circular` check from other projects' CI checks.

Prettier is not yet configured, as there are some incoming changes in https://github.com/Joystream/joystream/pull/945 and https://github.com/Joystream/joystream/pull/884 which would cause unnecessary conflicts.

In order to enable prettier we would just need to change `.prettierignore` file to:
```
**/*.js
**/*.d.ts
!.eslintrc.js
hiring/schemas/role.schema.json
lib/
build/
```
_(This is based on .gitignore, already tested locally)_

### CODE REVIEW

File: types/.gitignore [hunk 1]
```diff
@@ -1,6 +1,7 @@
 # Don't track build artifacts
 **/*.js
 **/*.d.ts
+!.eslintrc.js
```
Comment (mnaamani) [inline types/.gitignore:4]: what does the `!` here do?

**Review by mnaamani — COMMENTED** [review]
Looks good, left a question about what exclamation mark in .gititnore means

I ran yarn checks and it is reporting a circular dependency:

```
MokoBook:types mokhtar$ yarn checks
yarn run v1.22.4
$ yarn build && madge --circular ./ && yarn lint && prettier ./ --check
$ tsc --build tsconfig.json
Processed 73 files (866ms) 

✖ Found 2 circular dependencies!

1) lib/content-working-group/index.js > lib/index.js
2) lib/index.js > lib/members.js

error Command failed with exit code 1.
```

Although in the github checks it doesn't seem to give the same result

**Comment by mnaamani** [comment]
> Although in the github checks it doesn't seem to give the same result

Of course these were leftover artifacts from prior types library when output was being built into the `lib/` directory so this is a non issue

**Comment by mnaamani** [comment]
Can we add a step to do the following:

```
npm pack
tar xzf  joystream-types-0.11.0.tgz
cd pacakge/
npm install
node ./index.js
```

As a sanity check that the published package will work as expected

File: types/.gitignore (hunk 1 above)
Comment (Lezek123) [reply types/.gitignore:4]: It means this file is prevented from beeing ignored even if it's matched by some previous rule (`**/*.js` in this case)

Comment (mnaamani) [reply types/.gitignore:4]: ok

**Comment by Lezek123** [comment]
I think we would need a workaround that will work for all versions, so we don't have to change the version in `tar xzf  joystream-types-0.11.0.tgz` each time manually. I'll try to look into that.

**Comment by Lezek123** [comment]
Since the last line of output of `npm pack` should be the filename I think we can achieve this with:
`npm pack | tail -1 | xargs tar xzf`

**Comment by Lezek123** [comment]
Added `npm pack` CI check and prettier formatting

**Comment by mnaamani** [comment]
> Since the last line of output of `npm pack` should be the filename I think we can achieve this with:
> `npm pack | tail -1 | xargs tar xzf`

cool :+1:

**Pull request merged**